backend/src/dao/TeamEventAttendanceDao.ts

```diff
@@ -51,4 +52,59 @@ export default class TeamEventAttendanceDao {
     const docRef = teamEventAttendanceCollection.doc(uuid);
     await docRef.delete();
   }
+
+  /**
+   * Deletes all TEC Attendance
+   */
+  static async deleteAllTeamEventAttendance(): Promise<void> {
+    const batch = teamEventAttendanceCollection.firestore.batch();
+    const coll = await teamEventAttendanceCollection.get();
+
+    coll.docs.forEach((doc) => batch.delete(doc.ref));
+    await batch.commit();
```

Comment:
This logic looks mostly correct. Just want to double check with @henry-li-06 @JacksonStaniec though. Does the batch get all the documents within the collection, or just a selection of them? I wonder if we tried this on a larger amount of data (i.e. 500+ documents), if this would fail. 

Reading up [here](https://cloud.google.com/firestore/quotas#writes_and_transactions). 

> 

Maximum number of writes that can be passed to a Commit operation or performed in a transaction | 500
-- | --




If the batch ends up going over 500, this delete would fail, since we can't make more than 500 writes at once. Also, the code assumes that the doc is within the batch. What happens when we call `batch.delete(...)` and the document is not in the batch? We don't have a fallback for that.

This wouldn't really be a problem for a bit, especially since we're not going to use this method for awhile, but I can see us getting up to 500 fast. If you don't end up making any changes here, I would at the very least mention in the specification that this could fail if there's 500+ documents, and we'd file a ticket to come back and fix this.

I found some recipes here for deleting all documents in a collection, one of them takes into account the 500 limit and splits up the batches into max 500 documents each: 

https://firebase.google.com/docs/firestore/manage-data/delete-data#collections

https://stackoverflow.com/questions/48175235/how-to-delete-multiple-documents-from-cloud-firestore

Reply:
Ah, great @oscarwang20 ignore the solutions I posted then! Looks like we already have a solution (which happens to be taken directly off the Firebase solution I linked).